### PR TITLE
Tests for WDL functions in exciting locations

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/resources/standardTestCases/wdl_function_locations.test [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: wdl_function_locations
+testFormat: workflowsuccess
+ignore: true
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/wdl_function_locations.test:3]: why ignore: true?

**Comment by mcovarr** [comment]
I like what this is testing, but I'm not sure it needs to be a Centaur test.

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/65/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/wdl_function_locations.test (hunk 1 above)
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/wdl_function_locations.test:3]: Because it's currently broken :(

**Comment by cjllanwarne** [comment]
I personally think one of the main benefits of centaur is moving a lot of black-box "run a WDL; did it work" tests out into a parallelisable test framework.

Another benefit is that we can also catch cases where JES backend differs from Local.

**Comment by mcovarr** [comment]
Centaur is quite heavyweight and not part of what a developer typically does during the course of developing a new feature that could introduce regressions.  The only engine function referenced by this test is `range` whose implementation is not JES specific.

**Comment by cjllanwarne** [comment]
To me this is why the centaur as a github service should be way higher priority that it is currently.

We **must** be running centaur as part of our development process otherwise we **will** end up with regressions...

**Comment by geoffjentry** [comment]
I think this is a convo to have not in github but I agree with both of you.

I'm not a fan of unit tests that are comprised of "run a real workflow and look at some stuff" and think those should outside functional tests. OTOH centaur takes a while to run (OTOOH I think our unit tests take too long, primarily because of all of the functional tests happening in it). 

I agree w/ @mcovarr original statement although would prefer that it not be of the form of "run a wdl, and check it afterwards" but by actually testing the individual units 

Incentive to get Cromwell to scale better: It will allow centaur to be run fully parallel and thus only 5-10 mins long.

**Comment by kshakir** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/65/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Pull request merged**